### PR TITLE
Support Asynchronous and Synchronous request in bandsv

### LINKED ISSUE #431: add /proof/:requestId in bandsv
Implement `/proof/:requestId` route in bandsv

[specification here](https://documenter.getpostman.com/view/10283798/SWTEcbUC?version=latest)

### LINKED ISSUE #432: add type in post /request in bandsv
We need this endpoint return in 3 ways

1. `type == "UNCONFIRM"`, it will respond unconfirmed TxHash.
2. `type == "CONFIRM"`, it will respond confirmed TxHash.
3. `type == "PROOF"`, it will respond confirmed TxHash with proof.

[Specification](https://documenter.getpostman.com/view/10283798/SWTEcbUC?version=latest)

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #432 

What to do
- [x] Unconfirmed (ASYNCHRONOUS)
- [x] Confirmed (SYNCHRONOUS)
- [x] Confirmed + Proof (FULL)

### CODE REVIEW

File: chain/cmd/bandsv/main.go [hunk 1]
```diff
@@ -146,6 +169,26 @@ func handleRequestData(c *gin.Context) {
...
+			return
+		}
+
+		c.JSON(200, OracleRequestUnconfirmedResp{
+			TxHash: txr.TxHash,
+		})
+		return
+	}
+
+	fmt.Println("After unconfirmed")
```
Comment (taobun) [inline chain/cmd/bandsv/main.go]: If this line want to make other understand about code, make it to comment seem more make sense

File: chain/cmd/bandsv/main.go [hunk 2]
```diff
@@ -177,10 +220,14 @@ func handleRequestData(c *gin.Context) {
...
-		TxHash:    txr.TxHash,
-	})
+	if reqType == Confirmed {
+		c.JSON(200, OracleRequestConfirmedResp{
+			TxHash:    txr.TxHash,
+			RequestId: requestId,
+		})
+		return
+	}
+	fmt.Println("After confirmed")
```
Comment (taobun) [inline chain/cmd/bandsv/main.go]: This too

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go [hunk 3]
```diff
@@ -27,14 +27,31 @@ import (
 	"github.com/bandprotocol/d3n/chain/wasm"
 )
 
+const (
+	Unconfirmed = "UNCONFIRMED"
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Let's call this `ASYNCHRONOUS`, `SYNCHRONOUS`, and `FULL`

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by taobun — DISMISSED** [review]

File: chain/cmd/bandsv/main.go [hunk 4]
```diff
@@ -145,6 +168,25 @@ func handleRequestData(c *gin.Context) {
 
 	params := respParams.Result
 
+	// unconfirmed respond
+	if reqType == Asynchronous {
+		// TODO: Replace by `BandStatefulClient` after implementation finished.
+		txr, err := bandProvider.SendTransaction(
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: Do not duplicate code `txr, err := bandProvider.SendTransaction(...)`. Let's write it like this

```
txr, err := bandProvider.SendTransaction(...)
if err != nil {
     ...
}
if reqType == Asynchronous {
    ....
}
```

File: chain/cmd/bandsv/main.go [hunk 5]
```diff
@@ -26,14 +26,31 @@ import (
...
+type OracleRequestAsynchronousResp struct {
+	TxHash string `json:"txHash"`
+}
+
+type OracleRequestSynchronousResp struct {
 	TxHash    string `json:"txHash"`
+	RequestId uint64 `json:"id"`
+}
+
+type OracleRequestFullResp struct {
```
Comment (sorawit) [inline chain/cmd/bandsv/main.go]: I think we can combine the struct. 
```
type OracleRequestResp struct {
    TxHash string
    RequestId *uint64
    Proof ...
}
```
Then just set to nil if not applicable. Would that work @taobun ?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandsv/main.go (hunk 4 above)
Comment (evilpeach) [reply chain/cmd/bandsv/main.go]: In the case of Asynchronous, we use `flags.BroadcastAsync` and we use `flags.BroadcastBlock` in synchronous.

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**